tests/tests.rs

```diff
@@ -1401,3 +1343,279 @@ fn should_use_custom_response_size_estimate() {
         .expect_consistent();
     assert_matches!(response, Ok(_));
 }
+
+#[test]
+fn should_use_fallback_public_url() {
+    let authorized_caller = PrincipalId::new_user_test_id(ADDITIONAL_TEST_ID);
+    let setup = EvmRpcSetup::with_args(InitArgs {
+        demo: Some(true),
+        manage_api_keys: Some(vec![authorized_caller.0]),
+    });
+    let response = setup
+        .eth_get_transaction_count(
+            RpcServices::EthMainnet(Some(vec![EthMainnetService::Ankr])),
+            None,
+            candid_types::GetTransactionCountArgs {
+                address: "0xdAC17F958D2ee523a2206206994597C13D831ec7".to_string(),
+                block: candid_types::BlockTag::Latest,
+            },
+        )
+        .mock_http(
+            MockOutcallBuilder::new(200, r#"{"jsonrpc":"2.0","id":0,"result":"0x1"}"#)
+                .with_url("https://rpc.ankr.com/eth"),
+        )
+        .wait()
+        .expect_consistent()
+        .unwrap();
+    assert_eq!(response, 1);
+}
+
+#[test]
+fn should_update_api_key() {
+    let authorized_caller = PrincipalId::new_user_test_id(ADDITIONAL_TEST_ID);
+    let setup = EvmRpcSetup::with_args(InitArgs {
+        demo: Some(true),
+        manage_api_keys: Some(vec![authorized_caller.0]),
+    })
+    .as_caller(authorized_caller);
+    let provider_id = 1; // Ankr / mainnet
+    setup.update_api_keys(&[(provider_id, Some("test-api-key".to_string()))]);
```

Comment:
nit: it would help readability if `"test-api-key"` was extracted to a variable that's used here and on L. 1394 (in the mock) so that's it's apparent where the key is used. Same remark for the other tests.

Reply:
Sure; updated along with a few other similar changes throughout `tests.rs`.